### PR TITLE
Replace null characters in GitHub repo descriptions

### PR DESCRIPTION
**Pull request by pjlast**
GitHub repo descriptions can contain NULL characters, but Postgres does not support this for text fields. We now remove them before attempting to store the repo.

## Test plan

Verified manually by cloning https://github.com/Phr33d0m/NW-Profession-Bot , which has the following description:

```
"Automatically selects professions for empty slots\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00"
```

### CODE REVIEW

File: internal/repos/github.go [hunk 1]
```diff
@@ -412,6 +412,7 @@ func (s *GitHubSource) GetRepo(ctx context.Context, nameWithOwner string) (*type
 }
 
 func (s *GitHubSource) makeRepo(r *github.Repository) *types.Repo {
+	r.Description = strings.ReplaceAll(r.Description, "\x00", "") // Postgres does not support the NULL character in text fields
```
Comment (eseliger) [inline internal/repos/github.go:415]: a test case would be superb

**Review by eseliger — APPROVED** [review]
why why why would you allow null characters in text fields 😢 😆

File: internal/repos/github.go (hunk 1 above)
Comment (mrnugget) [reply internal/repos/github.go:415]: plus one

**Review by mrnugget — APPROVED** [review]

**Review by keegancsmith — COMMENTED** [review]
nice work. Should we go further and ensure the text is valid utf8 / remove non utf8 runs of chars with the empty string?

**Comment by pjlast** [comment]
@keegancsmith yeah that's probably a good idea. Do you perhaps know of an easy way to do this?

**Comment by keegancsmith** [comment]
A bit of a hack, but calling `strings.Map(func(r rune) rune { return r }, s)` should return a utf8 string with all invalid runes removed. If you want to replace bad runes with something else you should be able to check r against `utf8.RuneError`. There might be a better approach, but that one jumps to mind.

**Comment by keegancsmith** [comment]
haha, there is a function for you that does exactly whaty ou want. Don't use strings.Map. https://pkg.go.dev/strings#ToValidUTF8

**Pull request merged**